src/init.cpp

```diff
@@ -1343,6 +1344,13 @@ bool AppInitMain(boost::thread_group& threadGroup, CScheduler& scheduler)
             threadGroup.create_thread(&ThreadScriptCheck);
     }
 
+    sporkManager.InitDefaultSporkAddress();
+
+    if (IsArgSet("-sporkaddr")) {
+        if (!sporkManager.SetSporkAddress(GetArg("-sporkaddr", "")))
+            return InitError(_("Invalid spork address specified with -sporkaddr"));
+    }
+
```

Comment:
If you specify dummy (but valid e.g. the same as for testnet) `strSporkAddress` for regtest (in `chainparams.cpp`) then this can be simplified to

```
    if (!sporkManager.SetSporkAddress(GetArg("-sporkaddr", Params().SporkAddress())))
        return InitError(_("Invalid spork address specified with -sporkaddr"));
```

and `InitDefaultSporkAddress` can be dropped completely.

Reply:
What do you mean with dummy but valid? Is there any special meaning hidden in the testnet public key that I'm not aware of?

I was thinking of adding a new private/public key pair for regtest and adding a commit containing the private key in chainparams.cpp. What do you think?